### PR TITLE
Fixed bug when using a declarative function for the __tablename__

### PR DESCRIPTION
**Pull request by RmStorm**
When specifying a function to dynamically define a tablename as a declared
attribute it became impossible to overwrite it using a manually defined name.

Look at the test to see the issue, with the original code this test would fail.

### CODE REVIEW

File: gino/declarative.py [hunk 1]
```diff
@@ -166,8 +166,9 @@ def _init_table(cls, sub_cls):
                     updates[k] = sub_cls.__attr_factory__(k, v)
                 elif isinstance(v, (sa.Index, sa.Constraint)):
                     inspected_args.append(v)
-        if table_name is None:
-            table_name = getattr(sub_cls, '__tablename__', None)
+        if getattr(sub_cls, '__tablename__', None) and \
+                not callable(getattr(sub_cls, '__tablename__')):
+            table_name = getattr(sub_cls, '__tablename__')
```
Comment (wwwjfy) [inline gino/declarative.py]: Good catch!
But this still doesn't cover all cases, though most likely corner cases. We should move it into the loop.
Could you help to do that?

Comment (RmStorm) [reply gino/declarative.py]: Hi, I can. But which case isn't caught using this? I agree it's nicer to have it in the loop anyways, but I think it's probably good to write another test for any uncaught situation.

Comment (wwwjfy) [reply gino/declarative.py]: For example, one of the parent classes or mixins specify `__tablename__` as a variable

Comment (RmStorm) [reply gino/declarative.py]: Ah I see, that makes sense. I added a test for that case as well. Maybe I should also have added some tests that are supposed to throw errors on faulty behaviour... but I think it's good to go like this. Can this be merged and pushed to pypi? And maybe #590 can be merged as well?

**Comment by wwwjfy** [comment]
Thanks

**Pull request merged**